host/sgx/switchless.c

```diff
@@ -10,6 +11,68 @@
 #include "../ocalls.h"
 #include "enclave.h"
 
+#if __linux__
```

Comment:
rather than having #if _linux__ around Linux-specific code, it should be moved to a linux-specific file.   Just like we have host/optee/linux/enclave.c for the same reason, this should be in host/sgx/linux/switchless.c

Reply:
Hmm? It's already under the sgx directory, which limits us to just Linux and Windows? This means we would now have both "host/linux" and "host/sgx/linux" directories. I guess it makes sense.

Reply:
we already have both host/linux and host/optee/linux for exactly the same reason.

Reply:
Fixed. Added host/sgx/inux/switchless.c and host/sgx/windows/switchless.c